### PR TITLE
Validation and documentation for PodDiscriptionBudget change

### PR DESCRIPTION
**Pull request by chiayi**
**What type of PR is this?**
/kind cleanup

**What this PR does / Why we need it**:
This is additional cleanup for [PR #2740 ](https://github.com/googleforgames/agones/pull/2740). Added validation to return error when both minAvailable and maxUnavailable are both set.

### CODE REVIEW

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 1]
```diff
@@ -217,14 +217,14 @@ The following tables lists the configurable parameters of the Agones chart and t
...
+| Parameter                             | Description                                                                                                                                                                                                                           | Default |
+|---------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| `agones.allocator.pdb.enabled`        | Set to `true` to enable the creation of a [PodDisruptionBudget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) for the allocator deployment                                                                         | `false` |
+| `agones.allocator.pdb.minAvailable`   | Description of the number of pods from that set that must still be available after the eviction, even in the absence of the evicted pod. Can be either an absolute number or a percentage.                                            | `1`     |
```
Comment (gongmax) [inline site/content/en/docs/Installation/Install Agones/helm.md:223]: Need to add the statement here and the line below too?

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -217,14 +217,14 @@ The following tables lists the configurable parameters of the Agones chart and t
...
+| `agones.allocator.pdb.maxUnavailable` | Description of the number of pods from that set that can be unavailable after the eviction. It can be either an absolute number or a percentage.                                                                                      | \`\`    |
+| `agones.ping.pdb.enabled`             | Set to `true` to enable the creation of a [PodDisruptionBudget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) for the ping deployment                                                                              | `false` |
+| `agones.ping.pdb.minAvailable`        | Description of the number of pods from that set that must still be available after the eviction, even in the absence of the evicted pod. Can be either an absolute number or a percentage. **Mutually Exclusive with maxUnavailable** | `1`     |
```
Comment (zmerlynn) [inline site/content/en/docs/Installation/Install Agones/helm.md:226]: Super-nit here and below: Skip the bold, remember the code formatting and fix capitalization, i.e.:

Mutually exclusive with `maxUnavailable`.

**Review by zmerlynn — APPROVED** [review]

File: install/helm/agones/templates/ping.yaml [hunk 3]
```diff
@@ -97,6 +97,11 @@ kind: PodDisruptionBudget
 metadata:
   name: agones-ping-pdb
 spec:
+{{- if .Values.agones.ping.pdb.minAvailable }}
```
Comment (gongmax) [inline install/helm/agones/templates/ping.yaml:100]: Do we also want to add this validation to the `allocation.yaml`?

File: install/helm/agones/templates/service/allocation.yaml [hunk 4]
```diff
@@ -289,6 +289,11 @@ kind: PodDisruptionBudget
 metadata:
   name: agones-allocator-pdb
 spec:
+{{- if .Values.agones.ping.pdb.minAvailable }}
+{{- if .Values.agones.ping.pdb.maxUnavailable }}
```
Comment (gongmax) [inline install/helm/agones/templates/service/allocation.yaml:293]: Typo here. `ping` -> `allocator`.

Comment (chiayi) [reply install/helm/agones/templates/service/allocation.yaml]: Oops. Thanks.

**Review by gongmax — APPROVED** [review]

**Review by roberthbailey — APPROVED** [review]

**Pull request merged**